.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,46 @@
+<!--- Provide a general summary of your changes in the Title above -->
+
+<!--- Please make sure to create an issue for your task before creating -->
+<!--- a pull pequest. (Ideally, you should create the issue before you -->
+<!--- begin work, but better late than never.) This is how we keep track -->
+<!--- of the discussion around planned and in-development features and also -->
+<!--- how we can later find the code from the discussion and vice versa. -->
+
+<!--- Please complete the following sections with a level of detail -->
+<!--- determined at your discretion. In general, the amount of description -->
+<!--- should be roughly proportional to the size and significance of the changes. -->
+
+## Description
+<!--- Describe your changes in detail. Focus on the high-level "how" rather than -->
+<!--- the nitty-gritty "what"; i.e. avoid repeating the diff in prose. -->
+
+## Motivation and Context
+<!--- Why is this change required? What problem does it solve? -->
+<!--- If it fixes an open issue, please mention the issue here; -->
+<!--- e.g. "Closes #4." This also tells GitHub to automatically -->
+<!--- close the issue when the PR is merged. -->
+
+## How Has This Been Tested?
+<!--- Please describe in detail how you tested your changes. -->
+<!--- Include details of your testing environment, tests ran to see how -->
+<!--- your change affects other areas of the code, etc. -->
+
+## Screenshots (if appropriate):
+<!--- Showing off your awesome work is encouraged! -->
+<!--- You could also consider a quick demo via animated GIF, -->
+<!--- e.g. using https://giphy.com/apps/giphycapture -->
+<!--- Alternatively, a picture of a cute animal is always nice. -->
+
+## Types of changes
+<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+<!--- NOTE: breaking changes will probably be rare until we've amassed some functionality to break. -->
```

Comment:
May want to put in more details on breaking changes; e.g. 

> Since a breaking change requires an API version bump we try to avoid these whenever possible. Please explain here why the breaking change is required.